x-pack/plugins/ingest_manager/server/services/epm/packages/install.ts

```diff
@@ -188,16 +193,26 @@ export async function installPackage({
   // update current backing indices of each data stream
   await updateCurrentWriteIndices(callCluster, installedTemplates);
 
-  // if this is an update, delete the previous version's pipelines
-  if (installedPkg && !reinstall) {
+  // if this is an update or retrying an update, delete the previous version's pipelines
+  if (installType === InstallType.update || installType === InstallType.reupdate) {
     await deletePreviousPipelines(
       callCluster,
       savedObjectsClient,
       pkgName,
+      // @ts-ignore
       installedPkg.attributes.version
     );
   }
-
+  // pipelines from a different version may have installed during a failed update
+  if (installType === InstallType.rollback) {
+    await deletePreviousPipelines(
+      callCluster,
+      savedObjectsClient,
+      pkgName,
+      // @ts-ignore
```

Comment:
Please add a note about what we're ignoring if we can't avoid it

Reply:
see above